uui-core/src/data/processing/ItemsMap.ts

```diff
@@ -0,0 +1,96 @@
+export type OnUpdate<TId, TItem> = (newItemsMap: ItemsMap<TId, TItem>) => void;
+
+interface ModificationOptions {
```

Comment:
We can be more explicit in this API - just tell callee how we got these items:

```typescript
interface ModificationOptions {
  /** Where new/updated items came from:
  * api - received them from the recent api call
  * edit - received from dataRow.onValueChange callback
  */
  source: 'api' | 'edit'; 
}
```

Users can then decide what to do in each case:

```typescript
setItems((updatedItems, options) => {
   let items = new Map(myState.items);   
   if (options.source === 'api') {
      items = new Map([...updatedItems, ...items]) // items from state take precedence, to not override edited items
   } else {
      items = new Map([...items, ...updatedItem]) // items from state take precedence, to not override edited items
   }
})